src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -0,0 +1,326 @@
+            setTags(toCopy.tags);
+        }
+
+        /**
+         * Returns true if at least one field is edited.
+         */
+        public boolean isAnyFieldEdited() {
+            return CollectionUtil.isAnyNonNull(name, completionTime, servingSize, ingredients, steps, tags);
+        }
+
+        public void setName(Name name) {
+            this.name = name;
+        }
+
+        public Optional<Name> getName() {
+            return Optional.ofNullable(name);
+        }
+
+        public void setCompletionTime(CompletionTime completionTime) {
+            this.completionTime = completionTime;
+        }
+
+        public Optional<CompletionTime> getCompletionTime() {
+            return Optional.ofNullable(completionTime);
+        }
+
+        public void setServingSize(ServingSize servingSize) {
+            this.servingSize = servingSize;
+        }
+
+        public Optional<ServingSize> getServingSize() {
+            return Optional.ofNullable(servingSize);
+        }
+
+        public void setIngredients(List<Ingredient> ingredients) {
+            this.ingredients = ingredients;
+        }
+
+        public Optional<List<Ingredient>> getIngredients() {
+            return Optional.ofNullable(ingredients);
+        }
+
+        public void setSteps(List<Step> steps) {
+            this.steps = steps;
+        }
+
+        public Optional<List<Step>> getSteps() {
+            return Optional.ofNullable(steps);
+        }
+
+        /**
+         * Sets {@code tags} to this object's {@code tags}.
+         * A defensive copy of {@code tags} is used internally.
+         */
+        public void setTags(Set<Tag> tags) {
+            this.tags = (tags != null) ? new HashSet<>(tags) : null;
+        }
+
+        /**
+         * Returns an unmodifiable tag set, which throws {@code UnsupportedOperationException}
+         * if modification is attempted.
+         * Returns {@code Optional#empty()} if {@code tags} is null.
+         */
+        public Optional<Set<Tag>> getTags() {
+            return (tags != null) ? Optional.of(Collections.unmodifiableSet(tags)) : Optional.empty();
+        }
+
+        @Override
+        public boolean equals(Object other) {
+            // instanceof handles nulls
+            if (!(other instanceof EditRecipeDescriptor)) {
+                return false;
+            }
+
+            // short circuit if same object
+            if (other == this) {
+                return true;
+            }
+
+            // state check
+            EditRecipeDescriptor e = (EditRecipeDescriptor) other;
+
+            return getName().equals(e.getName())
+                    && getCompletionTime().equals(e.getCompletionTime())
+                    && getServingSize().equals(e.getServingSize())
+                    && getIngredients().equals(e.getIngredients())
+                    && getSteps().equals(e.getSteps())
+                    && getTags().equals(e.getTags());
+        }
+    }
```

Comment:
I think this can be abstracted out to a class on its own I'm not sure if it's good practice to include a class in a class.

Reply:
update: I just saw that you deleted the class for that and I wanted to ask what led to that decision?

Reply:
This was a copy pasta from AB3, can decide to move to another class if necessary but i think for now can leave it here